storage/database/sharded_database.go

```diff
@@ -151,102 +154,280 @@ func (pdb *shardedDB) Close() {
-	//			minIdx = idx
-	//			minIter = iter
-	//			minKey = iter.Key()
-	//			minKeyValue = iter.Value()
-	//		}
-	//	}
-	//}
-	//
-	//if minIter == nil {
-	//	return false
-	//}
-	//
-	//pdi.key = minKey
-	//pdi.value = minKeyValue
-	//
-	//if !minIter.Next() {
-	//	pdi.iterators[minIdx] = nil
-	//}
-	//
-	return true
-}
-
-func (pdi *shardedDBIterator) Error() error {
-	// TODO-Klaytn implement this later.
-	return nil
+func (it *shardedDBIterator) Key() []byte {
+	return it.key
 }
 
-func (pdi *shardedDBIterator) Key() []byte {
-	// TODO-Klaytn implement this later.
-	return nil
+func (it *shardedDBIterator) Value() []byte {
+	return it.value
 }
 
-func (pdi *shardedDBIterator) Value() []byte {
-	// TODO-Klaytn implement this later.
-	return nil
+func (it *shardedDBIterator) Release() {
+	it.cancel()
+}
+
+type entry struct {
+	key, val []byte
+}
+
+// shardedDBIteratorUnsorted iterates all items of each shardDB.
+// This is useful when you want to get items fast in serial.
+type shardedDBIteratorUnsorted struct {
+	shardedDBIterator
+}
+
+// NewIteratorUnsorted creates a iterator over the entire keyspace contained within
+// the key-value database.
+// If you want to get ordered items in serial, checkout shardedDB.NewIterator()
+// If you want to get items in parallel from channels, checkout shardedDB.NewChanIterator()
+func (db *shardedDB) NewIteratorUnsorted() Iterator {
+	return db.newIteratorUnsorted(func(db Database) Iterator { return db.NewIterator() })
+}
+
+// NewIteratorWithStartUnsorted creates a iterator over a subset of database content
+// starting at a particular initial key (or after, if it does not exist).
+func (db *shardedDB) NewIteratorWithStartUnsorted(start []byte) Iterator {
+	return db.newIteratorUnsorted(func(db Database) Iterator { return db.NewIteratorWithStart(start) })
+}
+
+// NewIteratorWithPrefixUnsorted creates a iterator over a subset of database content
+// with a particular key prefix.
+func (db *shardedDB) NewIteratorWithPrefixUnsorted(prefix []byte) Iterator {
+	return db.newIteratorUnsorted(func(db Database) Iterator { return db.NewIteratorWithPrefix(prefix) })
+}
+
+func (db *shardedDB) newIteratorUnsorted(newIterator func(Database) Iterator) Iterator {
+	it := &shardedDBIteratorUnsorted{
+		shardedDBIterator{
+			db.NewChanIterator(context.Background(), newIterator),
+			make(chan entry, shardedDBCombineChanSize),
+			nil, nil}}
+
+	go it.newCombineWorker()
+
+	return it
+}
+
+// runCombineWorker fetches any key/value from resultChs and put the data in resultCh
+func (it shardedDBIteratorUnsorted) newCombineWorker() {
+Iter:
+	for len(it.resultChs) != 0 {
+	chanIter:
+		for i, ch := range it.resultChs {
+			select {
+			case <-it.ctx.Done():
+				logger.Trace("[shardedDBIteratorUnsorted] combine worker ends due to ctx")
+				break Iter
+			case e, ok := <-ch:
+				if !ok { // channel is closed
+					it.resultChs = append(it.resultChs[:i], it.resultChs[i+1:]...)
+					break chanIter
+				} else {
```

Comment:
We don't need `else`.

Reply:
Thanks for the review. However, the code is deleted. There is no combine process and sub worker puts item in one channel. 😢 🙏